lib/commands/publish/npmPublishSemiAtomic.js

```diff
@@ -1,12 +1,19 @@
 var npmUpdateAsLatest = require("./npmUpdateAsLatest");
 var npmPublishAsTemp  = require("./npmPublishAsTemp");
+var execSync          = require("../../utils/execSync");
 
-module.exports = function npmPublishSemiAtomic(packages, packagesLoc, versions, callback) {
+module.exports = function npmPublishSemiAtomic(packages, packagesLoc, versions, flags, callback) {
   // Start by publishing a prerelease that we'll bump later to be more atomic
   npmPublishAsTemp(packages, packagesLoc, function (err) {
     if (err) return callback(err);
 
+    if (flags.canary) {
+      console.log("Reseting git state");
```

Comment:
Just commenting here if it was missed (the package.json files will be modified (dependencies and the version)), but we aren't commiting or creating a tag so the git state isn't empty so we reset.